# Carga inicial de los datos para explorar el contenido de cada uno de los archivos que componen los datasets

## Importación de librerías y configuraciones

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# ============================
# CONFIGURACIÓN DE PANDAS
# ============================

# Mostrar todas las columnas del DataFrame
pd.set_option('display.max_columns', None)

# Ajustar el ancho de la salida al tamaño de la ventana
pd.set_option('display.width', None)

# Mostrar el contenido completo de las celdas
pd.set_option('display.max_colwidth', None)

# Mostrar los números con dos decimales
pd.options.display.float_format = '{:,.2f}'.format

# ============================
# CONFIGURACIÓN DE GRÁFICOS
# ============================

# Tamaño por defecto de las figuras
plt.rcParams['figure.figsize'] = (10, 6)

# Estilo de los gráficos
sns.set_theme(style="whitegrid")

## Definición de rutas

In [ ]:
# Se importa pathlib para un uso mas eficiente de las rutas, en este caso creamos la ruta de la carpeta principal del proyecto

from pathlib import Path

BASE_DIR = Path.cwd().parent
BASE_DIR

PosixPath('/Users/luisamariamontealegrelopez/Desktop/ThePower/DataAnalitics/Proyecto_final')

In [6]:
# Ahora creamos una ruta para la carpeta de los datos originales uniendola con la carpeta principal

DATA_DIR = BASE_DIR / "Datos_originales"
DATA_DIR

PosixPath('/Users/luisamariamontealegrelopez/Desktop/ThePower/DataAnalitics/Proyecto_final/Datos_originales')

In [7]:
# Hacemos lo mismo con la carpeta ecommerce

ECOMMERCE_DIR = DATA_DIR / "ecommerce"
ECOMMERCE_DIR

PosixPath('/Users/luisamariamontealegrelopez/Desktop/ThePower/DataAnalitics/Proyecto_final/Datos_originales/ecommerce')

In [12]:
# Ahora con la carpeta marketing

MARKETING_DIR = DATA_DIR / "marketing"
MARKETING_DIR

PosixPath('/Users/luisamariamontealegrelopez/Desktop/ThePower/DataAnalitics/Proyecto_final/Datos_originales/marketing')

## Carga de datos

In [11]:
# Carga de los archivos del dataset ecommerce

customers = pd.read_csv(ECOMMERCE_DIR/"olist_customers_dataset.csv")
geolocation = pd.read_csv(ECOMMERCE_DIR/"olist_geolocation_dataset.csv")
order_items = pd.read_csv(ECOMMERCE_DIR/"olist_order_items_dataset.csv")
order_payments = pd.read_csv(ECOMMERCE_DIR/"olist_order_payments_dataset.csv")
order_reviews = pd.read_csv(ECOMMERCE_DIR/"olist_order_reviews_dataset.csv")
orders = pd.read_csv(ECOMMERCE_DIR/"olist_orders_dataset.csv")
products = pd.read_csv(ECOMMERCE_DIR/"olist_products_dataset.csv")
sellers = pd.read_csv(ECOMMERCE_DIR/"olist_sellers_dataset.csv")
product_category_name = pd.read_csv(ECOMMERCE_DIR/"product_category_name_translation.csv")

In [13]:
# Carga de los archivos del dataset marketing

marketing_qualified_leads = pd.read_csv(MARKETING_DIR / "olist_marketing_qualified_leads_dataset.csv")
closed_deals = pd.read_csv(MARKETING_DIR / "olist_closed_deals_dataset.csv")

## Exploración de los datos

# Dataset E-commerce

In [56]:
# Tabla customers
## 1. Dimensiones
customers.shape

(99441, 5)

In [57]:
## 2. Primeras filas
customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [58]:
## 3. Informacón general
customers.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   customer_id               99441 non-null  str  
 1   customer_unique_id        99441 non-null  str  
 2   customer_zip_code_prefix  99441 non-null  int64
 3   customer_city             99441 non-null  str  
 4   customer_state            99441 non-null  str  
dtypes: int64(1), str(4)
memory usage: 3.8 MB


In [59]:
## 4. Valores nulos

customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [60]:
## 5. Duplicados

customers.duplicated().sum()

np.int64(0)

In [61]:
## 6. Valores únicos en columnas customer_id y customer_unique_id

customers["customer_id"].nunique()

99441

In [62]:
customers["customer_unique_id"].nunique()

96096

Observaciones
El conjunto de datos contiene 99.441 registros y 5 variables.
No se identifican valores nulos ni registros duplicados completos.
Todas las variables presentan un tipo de dato adecuado para la información que almacenan.
customer_id es único para cada registro (99.441 valores únicos), por lo que identifica cada fila de la tabla.
customer_unique_id presenta 96.096 valores únicos, lo que indica que un mismo cliente puede aparecer asociado a más de un registro.

In [63]:
# Tabla orders
## 1. Dimensiones

orders.shape

(99441, 8)

In [64]:
## Información general

orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   order_id                       99441 non-null  str  
 1   customer_id                    99441 non-null  str  
 2   order_status                   99441 non-null  str  
 3   order_purchase_timestamp       99441 non-null  str  
 4   order_approved_at              99281 non-null  str  
 5   order_delivered_carrier_date   97658 non-null  str  
 6   order_delivered_customer_date  96476 non-null  str  
 7   order_estimated_delivery_date  99441 non-null  str  
dtypes: str(8)
memory usage: 6.1 MB


In [65]:
## 3. Valores nulos

orders.isna().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

In [66]:
## 4. Duplicados

orders.duplicated().sum()

np.int64(0)

In [67]:
## 5. Valores únicos en las claves

orders["order_id"].nunique()

99441

In [68]:
orders["customer_id"].nunique()

99441

In [69]:
orders.head()

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


In [70]:
## 7. Valores de la columna status

orders["order_status"].value_counts()

order_status
delivered      96478
shipped         1107
canceled         625
unavailable      609
invoiced         314
processing       301
created            5
approved           2
Name: count, dtype: int64

Observaciones
order_id es único para cada registro, por lo que cada fila representa un pedido. 
Todas las columnas relacionadas con fechas se encuentran almacenadas como texto (object). 
Se identifican valores nulos en algunas fechas del proceso logístico (order_approved_at, order_delivered_carrier_date y order_delivered_customer_date), por lo que será necesario analizar si estos nulos responden al estado del pedido antes de decidir cualquier tratamiento. 
La gran mayoría de los pedidos tienen estado delivered, mientras que el resto se distribuye entre estados como shipped, canceled, processing, invoiced, approved, created y unavailable. 

In [71]:
# Tabla order_items
## 1. Dimensiones
order_items.shape

(112650, 7)

In [72]:
## 2. Informaciñon general
order_items.info()

<class 'pandas.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  str    
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  str    
 3   seller_id            112650 non-null  str    
 4   shipping_limit_date  112650 non-null  str    
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), str(4)
memory usage: 6.0 MB


In [73]:
## 3. Valores nulos
order_items.isna().sum()

order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

In [74]:
## 4. Duplicados
order_items.duplicated().sum()

np.int64(0)

In [75]:
## 5. Claves
order_items["order_id"].nunique()

98666

In [76]:
order_items["product_id"].nunique()

32951

Observaciones
No se identifican valores nulos ni registros duplicados completos.
Las variables numéricas (price y freight_value) presentan un tipo de dato adecuado, mientras que shipping_limit_date deberá convertirse a tipo datetime.
El número de registros es superior al de la tabla orders, lo que indica que cada pedido puede contener varios productos. Esta tabla representa las líneas de detalle de cada pedido y no los pedidos en sí.
Se identifican 98.666 pedidos únicos, una cifra inferior al total de pedidos registrados en la tabla orders.
La tabla contiene el precio del producto, el coste del envío y el vendedor asociado a cada artículo vendido.

In [77]:
# Tabla order payments
## 1. Dimensiones
order_payments.shape

(103886, 5)

In [78]:
## 2. Información general
order_payments.info()

<class 'pandas.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  str    
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  str    
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), str(2)
memory usage: 4.0 MB


In [79]:
## 3. Valores nulos
order_payments.isna().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

In [80]:
## 4. Duplicados
order_payments.duplicated().sum()

np.int64(0)

In [81]:
## 5. Primeras filas
order_payments.head()

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [82]:
## 6. Valores únicos en las claves
order_payments["order_id"].nunique()

99440

In [83]:
## 7. Valores columna payment_type
order_payments["payment_type"].value_counts()

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

Observaciones
No se identifican valores nulos ni registros duplicados completos.
Todas las variables presentan un tipo de dato adecuado.
La tabla registra 99.440 pedidos únicos, prácticamente la totalidad de los pedidos existentes en la tabla orders.
El número de registros es superior al número de pedidos, lo que indica que un mismo pedido puede estar asociado a más de un registro de pago.
La tarjeta de crédito es el método de pago predominante, seguida por boleto, voucher y tarjeta de débito. Los registros con el valor not_defined son testimoniales (3 casos).

In [84]:
# Tabla order_reviews
## 1. Dimensiones
order_reviews.shape

(99224, 7)

In [85]:
## 2. Información general
order_reviews.info()

<class 'pandas.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   review_id                99224 non-null  str  
 1   order_id                 99224 non-null  str  
 2   review_score             99224 non-null  int64
 3   review_comment_title     11568 non-null  str  
 4   review_comment_message   40977 non-null  str  
 5   review_creation_date     99224 non-null  str  
 6   review_answer_timestamp  99224 non-null  str  
dtypes: int64(1), str(6)
memory usage: 5.3 MB


In [86]:
## 3. Valores nulos
order_reviews.isna().sum()

review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

In [87]:
## 4. Duplicados
order_reviews.duplicated().sum()

np.int64(0)

In [88]:
## 5. Valores únicos en las claves
order_reviews["order_id"].nunique()

98673

In [89]:
## 6. Valores 
order_reviews["review_score"].value_counts().sort_index()

review_score
1    11424
2     3151
3     8179
4    19142
5    57328
Name: count, dtype: int64

Observaciones
No se identifican registros duplicados completos.
Las columnas review_creation_date y review_answer_timestamp deberán convertirse al tipo datetime.
Las columnas review_comment_title y review_comment_message presentan un elevado número de valores nulos. No obstante, estos parecen responder a que los clientes valoraron el pedido sin dejar un comentario, por lo que no representan necesariamente un problema de calidad de los datos.
Se identifican 98.673 pedidos únicos, lo que indica que no todos los pedidos registrados en la plataforma recibieron una valoración por parte del cliente.
La distribución de review_score muestra un claro predominio de las valoraciones de 5 estrellas, lo que sugiere un alto nivel de satisfacción general de los clientes.

In [90]:
# Tabla products
## 1. Dimensiones
products.shape


(32951, 9)

In [91]:
## 2. Información general
products.info()

<class 'pandas.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  str    
 1   product_category_name       32341 non-null  str    
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), str(2)
memory usage: 2.3 MB


In [92]:
## 3. Valores nulos
products.isna().sum()

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

In [93]:
## 4. Duplicados
products.duplicated().sum()

np.int64(0)

In [94]:
## 5. Valores únicos en las claves
products["product_category_name"].nunique()

73

In [95]:
## 6. Primeras filas
products.head()

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.00,287.00,1.00,225.00,16.00,10.00,14.00
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.00,276.00,1.00,"1,000.00",30.00,18.00,20.00
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.00,250.00,1.00,154.00,18.00,9.00,15.00
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.00,261.00,1.00,371.00,26.00,4.00,26.00
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.00,402.00,4.00,625.00,20.00,17.00,13.00


Observaciones
No se identifican registros duplicados completos.
La columna product_id identifica de forma única cada producto del catálogo.
Existen 610 productos con información ausente en la categoría, longitud del nombre, longitud de la descripción y número de fotografías. Al coincidir exactamente en las mismas filas, estos valores nulos parecen corresponder a productos con información incompleta en origen.
Solo 2 productos presentan valores nulos en las variables físicas (peso y dimensiones), por lo que se trata de un problema puntual.
Se identifican 73 categorías de productos.
Algunas variables numéricas aparecen almacenadas como float debido a la presencia de valores nulos.

In [96]:
# Tabla sellers
## 1. Dimensiones
sellers.shape

(3095, 4)

In [97]:
## 2. Información general
sellers.info()

<class 'pandas.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   seller_id               3095 non-null   str  
 1   seller_zip_code_prefix  3095 non-null   int64
 2   seller_city             3095 non-null   str  
 3   seller_state            3095 non-null   str  
dtypes: int64(1), str(3)
memory usage: 96.8 KB


In [98]:
## 3. Valores nulos
sellers.isna().sum()

seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

In [99]:
## 4. Duplicados
sellers.duplicated().sum()

np.int64(0)

In [100]:
## 5. Valores únicos en las claves
sellers["seller_id"].nunique()

3095

In [101]:
## 6. Valores
sellers["seller_state"].value_counts()

seller_state
SP    1849
PR     349
MG     244
SC     190
RJ     171
RS     129
GO      40
DF      30
ES      23
BA      19
CE      13
PE       9
PB       6
RN       5
MS       5
MT       4
RO       2
SE       2
AC       1
PI       1
MA       1
AM       1
PA       1
Name: count, dtype: int64

Observaciones
No se identifican valores nulos ni registros duplicados completos.
seller_id identifica de forma única a cada vendedor del marketplace.
Todas las variables presentan un tipo de dato adecuado.
La distribución geográfica de los vendedores es desigual.
Esta tabla permitirá incorporar información geográfica de los vendedores al dataset final y realizar análisis por ubicación, volumen de ventas y rendimiento.

In [102]:
# Tabla geolocation
## 1. Dimensiones
geolocation.shape

(1000163, 5)

In [103]:
## 2. Información general
geolocation.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  str    
 4   geolocation_state            1000163 non-null  str    
dtypes: float64(2), int64(1), str(2)
memory usage: 38.2 MB


In [104]:
## 3. Valores nulos
geolocation.isna().sum()

geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

In [105]:
## 4. Duplicados
geolocation.duplicated().sum()

np.int64(261831)

In [ ]:
## 5. Valores únicos
geolocation["geolocation_zip_code_prefix"].nunique()

19015

Observaciones
El conjunto de datos contiene 1.000.163 registros y 5 variables, siendo la tabla de mayor tamaño.
No presenta valores nulos.
Se identifican 261.831 registros duplicados completos.
La tabla contiene únicamente 19.015 códigos postales únicos, por lo que un mismo código postal aparece asociado a múltiples registros geográficos.
geolocation_zip_code_prefix no constituye una clave primaria, por lo que la tabla no puede integrarse directamente con el resto del modelo sin una transformación previa.
Será necesario agregar la información geográfica por código postal antes de realizar los merge, evitando así la duplicación masiva de registros en el dataset final.

In [107]:
# Tabla product_category_name
## 1. Dimensiones
product_category_name.shape

(71, 2)

In [108]:
## 2. Información general
product_category_name.info()

<class 'pandas.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   product_category_name          71 non-null     str  
 1   product_category_name_english  71 non-null     str  
dtypes: str(2)
memory usage: 1.2 KB


In [109]:
## 3. Valores nulos
product_category_name.isna().sum()

product_category_name            0
product_category_name_english    0
dtype: int64

In [110]:
## 4. Duplicados
product_category_name.duplicated().sum()

np.int64(0)

Observaciones
El conjunto de datos no presenta valores nulos ni registros duplicados.
La tabla actúa como una tabla de correspondencia entre el nombre original de las categorías de producto en portugués y su traducción al inglés.
No requiere transformaciones previas.
Será necesario comprobar durante la integración de los datos que todas las categorías presentes en products tienen una traducción asociada.

# Dataset marketing

In [111]:
# Tabla marketing qualified leads
## 1. Dimensiones
marketing_qualified_leads.shape

(8000, 4)

In [112]:
## 2. Información general
marketing_qualified_leads.info()

<class 'pandas.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   mql_id              8000 non-null   str  
 1   first_contact_date  8000 non-null   str  
 2   landing_page_id     8000 non-null   str  
 3   origin              7940 non-null   str  
dtypes: str(4)
memory usage: 250.1 KB


In [113]:
## 3. Valores nulos
marketing_qualified_leads.isna().sum()

mql_id                 0
first_contact_date     0
landing_page_id        0
origin                60
dtype: int64

In [114]:
## 4. Duplicados
marketing_qualified_leads.duplicated().sum()

np.int64(0)

In [116]:
## 5. Valores
marketing_qualified_leads["origin"].value_counts(dropna=False)

origin
organic_search       2296
paid_search          1586
social               1350
unknown              1099
direct_traffic        499
email                 493
referral              284
other                 150
display               118
other_publicities      65
NaN                    60
Name: count, dtype: int64

In [117]:
## 6. Primeras filas
marketing_qualified_leads.head()

,mql_id,first_contact_date,landing_page_id,origin
0,dac32acd4db4c29c230538b72f8dd87d,2018-02-01,88740e65d5d6b056e0cda098e1ea6313,social
1,8c18d1de7f67e60dbd64e3c07d7e9d5d,2017-10-20,007f9098284a86ee80ddeb25d53e0af8,paid_search
2,b4bc852d233dfefc5131f593b538befa,2018-03-22,a7982125ff7aa3b2054c6e44f9d28522,organic_search
3,6be030b81c75970747525b843c1ef4f8,2018-01-22,d45d558f0daeecf3cccdffe3c59684aa,email
4,5420aad7fec3549a85876ba1c529bd84,2018-02-21,b48ec5f3b04e9068441002a19df93c6c,organic_search


In [135]:
## 7. Valores únicos en las claves
marketing_qualified_leads["mql_id"].nunique()

8000

Observaciones
No se identifican registros duplicados completos y mql_id identifica de forma única cada lead.
La columna first_contact_date deberá convertirse al tipo datetime.
Solo se detectan 60 valores nulos en la variable origin, lo que representa una proporción muy reducida del conjunto de datos.
Los principales canales de captación son organic_search, paid_search y social, concentrando la mayor parte de los leads registrados.
Esta tabla permitirá analizar el origen de los leads y, junto con closed_deals, estudiar el rendimiento y la conversión de los distintos canales de marketing.

In [118]:
# Tabla closed deals
## 1. Dimensiones
closed_deals.shape

(842, 14)

In [119]:
## 2. Información general
closed_deals.info()

<class 'pandas.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    str    
 1   seller_id                      842 non-null    str    
 2   sdr_id                         842 non-null    str    
 3   sr_id                          842 non-null    str    
 4   won_date                       842 non-null    str    
 5   business_segment               841 non-null    str    
 6   lead_type                      836 non-null    str    
 7   lead_behaviour_profile         665 non-null    str    
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     str    
 11  business_type                  832 non-null    str    
 12  declared_product_catalog_size  69 non-null     float64
 13  d

In [120]:
## 3. Valores nulos
closed_deals.isna().sum()

mql_id                             0
seller_id                          0
sdr_id                             0
sr_id                              0
won_date                           0
business_segment                   1
lead_type                          6
lead_behaviour_profile           177
has_company                      779
has_gtin                         778
average_stock                    776
business_type                     10
declared_product_catalog_size    773
declared_monthly_revenue           0
dtype: int64

In [121]:
## 4. Duplicados
closed_deals.duplicated().sum()

np.int64(0)

In [125]:
## 5. Valores únicos en las claves
closed_deals["mql_id"].nunique()

842

In [126]:
closed_deals["seller_id"].nunique()

842

In [127]:
closed_deals["sdr_id"].nunique()

32

In [128]:
closed_deals["sr_id"].nunique()

22

In [129]:
## 6. Primeras filas
closed_deals.head()

,mql_id,seller_id,sdr_id,sr_id,won_date,business_segment,lead_type,lead_behaviour_profile,has_company,has_gtin,average_stock,business_type,declared_product_catalog_size,declared_monthly_revenue
0,5420aad7fec3549a85876ba1c529bd84,2c43fb513632d29b3b58df74816f1b06,a8387c01a09e99ce014107505b92388c,4ef15afb4b2723d8f3d81e51ec7afefe,2018-02-26 19:58:54,pet,online_medium,cat,NaN,NaN,NaN,reseller,NaN,0.00
1,a555fb36b9368110ede0f043dfc3b9a0,bbb7d7893a450660432ea6652310ebb7,09285259593c61296eef10c734121d5b,d3d1e91a157ea7f90548eef82f1955e3,2018-05-08 20:17:59,car_accessories,industry,eagle,NaN,NaN,NaN,reseller,NaN,0.00
2,327174d3648a2d047e8940d7d15204ca,612170e34b97004b3ba37eae81836b4c,b90f87164b5f8c2cfa5c8572834dbe3f,6565aa9ce3178a5caf6171827af3a9ba,2018-06-05 17:27:23,home_appliances,online_big,cat,NaN,NaN,NaN,reseller,NaN,0.00
3,f5fee8f7da74f4887f5bcae2bafb6dd6,21e1781e36faf92725dde4730a88ca0f,56bf83c4bb35763a51c2baab501b4c67,d3d1e91a157ea7f90548eef82f1955e3,2018-01-17 13:51:03,food_drink,online_small,NaN,NaN,NaN,NaN,reseller,NaN,0.00
4,ffe640179b554e295c167a2f6be528e0,ed8cb7b190ceb6067227478e48cf8dde,4b339f9567d060bcea4f5136b9f5949e,d3d1e91a157ea7f90548eef82f1955e3,2018-07-03 20:17:45,home_appliances,industry,wolf,NaN,NaN,NaN,manufacturer,NaN,0.00


In [130]:
## ültimas filas
closed_deals.tail()

,mql_id,seller_id,sdr_id,sr_id,won_date,business_segment,lead_type,lead_behaviour_profile,has_company,has_gtin,average_stock,business_type,declared_product_catalog_size,declared_monthly_revenue
837,df9a2abe2ad3f2d8758b675ac8331ecf,bfcc27719640628da877db48b672b169,45749fb708130f78d0db07d8d80f030b,a8387c01a09e99ce014107505b92388c,2018-08-01 20:22:31,home_decor,offline,cat,NaN,NaN,NaN,reseller,NaN,0.00
838,8723201570415eb23b4f095d8c20a27c,4a82eab98441aeb64566e2776c1fb2b6,370c9f455f93a9a96cbe9bea48e70033,4b339f9567d060bcea4f5136b9f5949e,2018-09-27 18:58:41,construction_tools_house_garden,online_big,wolf,NaN,NaN,NaN,reseller,132.00,"200,000.00"
839,2e57665b8faf05c967a801eb5aedfa0a,55031883943971ca22db6894574cfe2c,370c9f455f93a9a96cbe9bea48e70033,a8387c01a09e99ce014107505b92388c,2018-06-06 20:15:26,computers,online_medium,wolf,NaN,NaN,NaN,reseller,NaN,0.00
840,40955be51cc85c5a5a6cdd12b19e9f10,c0e933c238e41f0cd459d6025ee9b364,068066e24f0c643eb1d089c7dd20cd73,d3d1e91a157ea7f90548eef82f1955e3,2018-03-06 15:52:23,pet,online_beginner,cat,NaN,NaN,NaN,reseller,NaN,0.00
841,8a6492305a5fbcdcdd1a7f5a90764c07,1bba9c94eb5c4e82fba9f7286fe81025,56bf83c4bb35763a51c2baab501b4c67,de63de0d10a6012430098db33c679b0b,2018-05-02 13:31:32,household_utilities,industry,eagle,NaN,NaN,NaN,manufacturer,NaN,0.00


In [131]:
## 7. Contar valores
closed_deals["business_segment"].value_counts()

business_segment
home_decor                         105
health_beauty                       93
car_accessories                     77
household_utilities                 71
construction_tools_house_garden     69
audio_video_electronics             64
computers                           34
pet                                 30
food_supplement                     28
food_drink                          26
sports_leisure                      25
bed_bath_table                      22
bags_backpacks                      22
toys                                20
fashion_accessories                 19
home_office_furniture               14
stationery                          13
phone_mobile                        13
small_appliances                    12
handcrafted                         12
baby                                10
music_instruments                    9
books                                9
watches                              8
jewerly                              8
home_app

In [132]:
closed_deals["lead_type"].value_counts()

lead_type
online_medium      332
online_big         126
industry           123
offline            104
online_small        77
online_beginner     57
online_top          14
other                3
Name: count, dtype: int64

In [133]:
closed_deals["lead_behaviour_profile"].value_counts()

lead_behaviour_profile
cat            407
eagle          123
wolf            95
shark           24
cat, wolf        8
eagle, wolf      3
eagle, cat       3
shark, cat       1
shark, wolf      1
Name: count, dtype: int64

In [134]:
closed_deals["business_type"].value_counts()

business_type
reseller        587
manufacturer    242
other             3
Name: count, dtype: int64

Observaciones
mql_id identifica de forma única cada acuerdo cerrado y permite relacionar esta tabla con marketing_qualified_leads.
La columna won_date deberá convertirse al tipo datetime.
Se detectan pocos valores nulos en business_segment, lead_type y business_type, mientras que variables como has_company, has_gtin, average_stock y declared_product_catalog_size presentan un porcentaje muy elevado de valores ausentes, por lo que será necesario evaluar su utilidad antes de decidir si conservarlas.
Los segmentos de negocio con mayor representación son home_decor, health_beauty y car_accessories.
Predominan los leads de tipo online_medium, seguidos por online_big e industry.
La mayor parte de los acuerdos corresponden a empresas clasificadas como reseller, seguidas por manufacturer.
Esta tabla constituye el enlace entre el proceso de captación de leads y la incorporación de nuevos vendedores al marketplace.